.github/workflows/promote-charms.yaml

```
@@ -37,8 +37,11 @@ jobs:
           BRANCH=${BRANCH#refs/heads/}  # strip off refs/heads/ if it exists
           if [[ "${BRANCH}" == "main" ]]; then
             echo "track=latest" >> "$GITHUB_OUTPUT"
-          elif [[ "${BRANCH}" =~ "^release-[0-9]+\.[0-9]+$" ]]; then
+          elif [[ "${BRANCH}" =~ ^release-[0-9]+\.[0-9]+$ ]]; then
             echo "track=${BRANCH:8}" >> "$GITHUB_OUTPUT"
+          else
+            echo "::error Failed to determine track from branch ${BRANCH}"
+          exit 1
```

Perhaps another syntax error?
```suggestion
          else
            echo "::error Failed to determine track from branch ${BRANCH}"
            exit 1
          fi
```